### PR TITLE
Ingawei/add your groups to group page

### PR DESCRIPTION
**Pull request by ingawei**
@jahooma pls review

Separated groups page into two things, your groups and discover:
image
image

Also updated groups search to hopefully be more efficient and good (does the same thing as contract search, checks exact text and then does fuzzy search. The API calls are on dev if you'd like to try it out

### CODE REVIEW

File: web/lib/supabase/groups.ts [hunk 1]
```diff
@@ -124,3 +203,37 @@ export async function getPublicGroups() {
...
+
+export async function listGroupsBySlug(groupSlugs: string[]) {
+  const { data } = await run(
+    db.from('groups').select('data').in('slug', groupSlugs)
+  )
+  if (data && data.length > 0) {
+    return data.map((group) => group.data as Group)
+  }
+  return []
+}
```
Comment (jahooma) [inline web/lib/supabase/groups.ts:239]: I think @IanPhilips just added a few indexes for groups and related tables. Would be good to still check that these queries are covered! (Check seed.sql or see if your db tools shows what indexes were created, or run the query in your console with explain analyze and check it is < 40ms.)

File: web/pages/groups.tsx [hunk 2]
```diff
@@ -89,29 +49,22 @@ export default function Groups(props: { groups: SearchGroupInfo[] }) {
...
+              className={'mb-4'}
+              tabs={[
+                {
+                  title: 'Your Groups',
+                  content: <YourGroups yourGroupIds={yourGroupIds} />,
+                },
+                {
+                  title: 'Discover',
+                  content: <DiscoverGroups yourGroupIds={yourGroupIds} />,
+                },
```
Comment (jahooma) [inline web/pages/groups.tsx:63]: For users that are not in any groups, what do you think about defaulting to the Discover groups tab?

**Review by jahooma — APPROVED** [review]
This is great, thanks Inga! Glad to see this overhaul of the groups page, including proper search!

Code looks good.

Will be excited to see more usage of groups. I think we should include them in the feed cards, (but maybe after we overhaul the feed to be more like a timeline, an idea we had today).

File: web/pages/groups.tsx (hunk 2 above)
Comment (ingawei) [reply web/pages/groups.tsx:63]: ohh smart, will do

File: backend/api/src/supabase-search-groups.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      query = `
+      select groupz.data
+      from (
+        select groups.*,
+            similarity(groups.name,$1) AS similarity_score
+        FROM groups 
+        join group_members 
+        on groups.id = group_members.group_id
+        where group_members.member_id = '${uid}'
+      ) AS groupz
```
Comment (IanPhilips) [inline backend/api/src/supabase-search-groups.ts:85]: I sort of wish these `groupz` variables had better names, maybe like `similar_groups`?

File: backend/supabase/seed.sql [hunk 4]
```diff
@@ -620,7 +620,9 @@ create table if not exists
     privacy_status text,
     slug text,
     name text,
+    name_fts tsvector generated always as (to_tsvector('english'::regconfig, name)) stored,
     creator_id text;
```
Comment (IanPhilips) [inline backend/supabase/seed.sql:624]: should be a `,`, not `;`

File: backend/api/src/supabase-search-groups.ts [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+      `
+    } else {
+      query = `
+      select groups.data
+        from groups 
+        join group_members 
+        on groups.id = group_members.group_id,
+        websearch_to_tsquery('english',  $1) as query
+      where group_members.member_id = '${uid}'
+      and groups.name_fts @@ query
```
Comment (IanPhilips) [inline backend/api/src/supabase-search-groups.ts:97]: I don't see the `groups.name_fts` column in the prod db though it is in the seed.sql, you'll have to add it manually via 
`alter table groups
add column name_fts tsvector generated always as (to_tsvector(name)) stored;`

File: backend/supabase/groups/indexes.sql [hunk 6]
```diff
@@ -0,0 +1,5 @@
+
+create index if not exists group_slug on public.groups (slug);
+create index if not exists group_name on public.groups (name);
+create index if not exists group_creator_id on public.groups (creator_id);
+create index if not exists total_members on public.groups (total_members desc);
```
Comment (IanPhilips) [inline backend/supabase/groups/indexes.sql:5]: It looks like you're searching `name_fts` in non-fuzzy search, here's the query plan using the dev db:
Screenshot 2023-05-16 at 9 18 19 AM

File: backend/supabase/groups/indexes.sql [hunk 7]
```diff
@@ -0,0 +1,5 @@
+
+create index if not exists group_slug on public.groups (slug);
+create index if not exists group_name on public.groups (name);
```
Comment (IanPhilips) [inline backend/supabase/groups/indexes.sql:3]: Nice, looks like this index is working:
Screenshot 2023-05-16 at 9 21 10 AM

**Pull request merged**